### DISCUSSION TITLE
Cannot write an F-bounded type parameter bound: Cyclic reference involving type T

### DISCUSSION THREAD

**Discussion by neko-kai** [scala/scala3#17797] [Feature Requests]
## Compiler version

3.1.0

## Minimized code

```Scala
def xa[T[x <: T[x]]] = ()
```

## Output

```scala
Cyclic reference involving type T
```

## Expectation

Expected to work, as in Scala 2

**Comment by odersky** [comment]
Note: F-bounds are not expected to work as in Scala 2. We accept that they will produce cyclic references more often. It's a price to pay going from `with`-types to true intersections.
